examples/avro-cli.py

```diff
@@ -152,21 +176,21 @@ def consume(topic, conf):
 
 
 def main(args):
-    # handle common configs
-    conf = {'bootstrap.servers': args.bootstrap_servers,
-            'schema.registry.url': args.schema_registry}
-
+    registry_client = SchemaRegistryClient(url=args.schema_registry)
     if args.userinfo:
-        conf['schema.registry.basic.auth.credentials.source'] = 'USER_INFO'
-        conf['schema.registry.basic.auth.user.info'] = args.userinfo
+        registry_client.user_auth(args.userinfo.split(':'))
+
+    avro_serializer = AvroSerializer(registry_client, record_schema,
+                                     converter=UserConverter)
 
+    conf = {'bootstrap.servers': args.bootstrap_servers}
     if args.mode == "produce":
-        produce(args.topic, conf)
+        produce(args.topic, conf, avro_serializer)
```

Comment:
fwiw, when I read this my first thought was: 'why pass avro_serializer into the produce method?'.. so maybe it's better to call this `produce_msgs` or something, but v. minor point, if there is a point here at all.

Reply:
that's fair I should qualify all kwargs as well